drivers/cpl/nuopc/glc_comp_nuopc.F90

```diff
@@ -0,0 +1,934 @@
+          runtype='branch'
+       else
+          call shr_sys_abort(subname//' ERROR: unknown starttype' )
+       end if
+    else
+       if (my_task == master_task) then
+          write(stdout,*)' GLC cism is not evolving, runtype is always set to initial'
+          runtype = 'initial'
+       end if
+    end if
+
+    !----------------------
+    ! Get properties from clock
+    !----------------------
+
+    call ESMF_ClockGet( clock, &
+         currTime=currTime, startTime=startTime, stopTime=stopTime, timeStep=timeStep, rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    call ESMF_TimeGet( startTime, yy=yy, mm=mm, dd=dd, s=start_tod, rc=rc )
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    call shr_cal_ymd2date(yy,mm,dd,start_ymd)
+
+    call ESMF_TimeGet( currTime, yy=yy, mm=mm, dd=dd, s=curr_tod, rc=rc )
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    call shr_cal_ymd2date(yy,mm,dd,curr_ymd)
+
+    call ESMF_TimeGet( stopTime, yy=yy, mm=mm, dd=dd, s=stop_tod, rc=rc )
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    call shr_cal_ymd2date(yy,mm,dd,stop_ymd)
+
+    call ESMF_TimeGet( currTime, calkindflag=esmf_caltype, rc=rc )
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    if (esmf_caltype == ESMF_CALKIND_NOLEAP) then
+       ! do nothing
+    else if (esmf_caltype == ESMF_CALKIND_GREGORIAN) then
+       ! do nothing
+    else
+       call shr_sys_abort( subname//'ERROR:: bad calendar for ESMF' )
+    end if
+
+    !----------------------
+    ! Initialize GLC
+    !----------------------
+
+    ! now initialize GLC
+    call glc_initialize(clock)
+    if (my_task == master_task) then
+       write(stdout,F01) ' GLC Initial Date ',iyear,imonth,iday,ihour,iminute,isecond
+       write(stdout,F00) ' Initialize Done'
+    endif
+
+    ! TODO (mvertens, 2018-11-28): Determine if land is present as a sanity check - do we need this?
+    ! TODO (mvertens, 2018-11-28): read in model_doi_url
+
+    !--------------------------------
+    ! Generate the mesh on the cism decomposition
+    !--------------------------------
+
+    ! read in the mesh
+    call NUOPC_CompAttributeGet(gcomp, name='mesh_glc', value=cvalue, rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    if (my_task == master_task) then
+       write(stdout,*)'mesh file for domain is ',trim(cvalue)
+    end if
+
+    ! create distGrid from global index array
+    gindex = local_to_global_indices()
+
+    DistGrid = ESMF_DistGridCreate(arbSeqIndexList=gindex, rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    ! read in the mesh
+    EMesh = ESMF_MeshCreate(filename=trim(cvalue), fileformat=ESMF_FILEFORMAT_ESMFMESH, &
+          elementDistgrid=Distgrid,  rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    ! elemAreaArray = ESMF_ArrayCreate(DistGrid, mesh_areas, rc=rc)
+    ! if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    ! call ESMF_MeshGet(Emesh, elemAreaArray=elemAreaArray, rc=rc)
+    ! if (ChkErr(rc,__LINE__,u_FILE_u)) return
```

Comment:
- [ ] @mvertens - should we remove this commented-out code, or will you be making use of it soon?